packages/authorization-platformstate-writer/src/consumerServiceV2.ts

```diff
@@ -360,20 +361,77 @@ export async function handleMessageV2(
         logger
       );
     })
+    .with({ type: "ClientAdminRemoved" }, async (msg) => {
+      const client = parseClient(msg.data.client, msg.type);
+      const pk = makePlatformStatesClientPK(client.id);
+      const clientEntry = await readPlatformClientEntry(pk, dynamoDBClient);
+
+      if (clientEntry) {
+        if (clientEntry.version > msg.version) {
+          logger.info(
+            `Skipping processing of entry ${clientEntry.PK}. Reason: a more recent entry already exists`
+          );
+          return Promise.resolve();
+        } else {
```

Comment:
Using if-else is generally not recommended, as it can be harder to maintain.
Wouldn't a solution like this be more readable?

```suggestion
    if (!clientEntry) {
      logger.info(
        'Skipping processing of ClientAdminRemoved event: client data is missing or has an invalid format.'
  );
  return Promise.resolve();
}

if (clientEntry.version > msg.version) {
  logger.info(
    `Skipping processing of entry ${clientEntry.PK}. Reason: a newer version already exists.`
  );
  return Promise.resolve();
}
```